docs/book/features/block_volume.md

```diff
@@ -0,0 +1,278 @@
+2. Static Volume Provisioning
+
+## Dynamic Volume Provisioning
+
+  Dynamic volume provisioning allows storage volumes to be created on-demand.
+
+  Without dynamic provisioning, cluster administrators have to manually make calls to their cloud or storage provider to
+  create new storage volumes and then create PersistentVolume objects to represent them in Kubernetes.
+
+  The dynamic provisioning feature eliminates the need for cluster administrators to pre-provision storage. Instead, it
+  automatically provisions storage when it is requested by users.
+
+  The implementation of dynamic volume provisioning is based on the API object `StorageClass` from the API group `storage.k8s.io`
+  A cluster administrator can define as many `StorageClass` objects as needed, each specifying a volume plugin
+  (a.k.a `provisioner`) that provisions a volume and a set of parameters to that `provisioner` when provisioning.
+  A cluster administrator can define and expose multiple flavors of storage (from the same or different storage systems)
+  within a cluster, each with a custom set of parameters.
+
+  Dynamic Volume Provisioning is supported both in Vanilla Kubernetes clusters and Project Pacific clusters.
+
+### Steps to Dynamically Provision a Block Volume on Vanilla Kubernetes Cluster
+
+  This section describes the step by step instructions to provision a PersistentVolume dynamically on a `Vanilla Kubernetes` Cluster
+
+* Define a Storage Class as shown [here](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/example/example-sc.yaml)
+
+* Import this `StorageClass` into `Vanilla Kubernetes` Cluster:
+
+```text
+    kubectl create -f example-sc.yaml
+```
+
+* Define a `PersistentVolumeClaim` request as shown in the spec [here](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/example/example-pvc.yaml)
+
+* Import this `PersistentVolumeClaim` into `Vanilla Kubernetes` Cluster:
+
+```text
+    kubectl create -f example-pvc.yaml
+```
+
+* Verify the PersistentVolumeClaim was created:
+
+  Check to see if the `PersistentVolumeClaim` we just imported was created and has a `PersistentVolume` attached to it.
+
+  The `Status` section below should show `Bound` if it worked and the `Volume` field should be populated.
+
+  A `PersistentVolume` is automatically created and is bound to this `PersistentVolumeClaim`.
+
+```text
+  $ kubectl describe pvc example-vanilla-block-pvc
+    Name:          example-vanilla-block-pvc
+    Namespace:     default
+    StorageClass:  example-vanilla-block-sc
+    Status:        Bound
+    Volume:        pvc-7ed39d8e-7896-11ea-a119-005056983fec
+    Labels:        <none>
+    Annotations:   pv.kubernetes.io/bind-completed: yes
+                   pv.kubernetes.io/bound-by-controller: yes
+                   volume.beta.kubernetes.io/storage-provisioner: csi.vsphere.vmware.com
+    Finalizers:    [kubernetes.io/pvc-protection]
+    Capacity:      5Gi
+    Access Modes:  RWO
```

Comment:
Can you also mention somewhere how  `Access Modes` is determining type of the volume to be provisioned?
With `RWM`, file volume will get provisioned and for block volume this mode has to be `RWO`.